migrations/src/main/scala/za/co/absa/enceladus/migrations/framework/ObjectIdTools.scala

```diff
@@ -79,4 +79,26 @@ object ObjectIdTools {
       document
     }
   }
+
+  /**
+    * Extracts an object id value from an object id json formatted object id.
+    * Here is an example. For the following input (all quotes are parts of the string itself):
+    *
+    * `"_id" : { "$oid" : "5b98eea5a43a28a6154a2453" }"`
+    *
+    * the output of the method will be:
+    *
+    * `5b98eea5a43a28a6154a2453`
+    *
+    * @param objIdExpression An Object Id as a JSON key-value expression.
+    * @return A string value of the Object Id
+    */
+  def extractId(objIdExpression: String): String = {
+    val idRegexp = """.*\"\$oid\"\s*:\s*\"(.*)\".*""".r
```

Comment:
I am thinking if `.*` is not too permissive, it's a hex and shouldn't be anything else.